File: docs/csharp/scripts/control-flow.cs
#:sdk Microsoft.NET.Sdk
// Princípios: guard clauses, retorno antecipado, switch expressions

Console.WriteLine(ProcessOrder(null));
Console.WriteLine(ProcessOrder(new Order(1, 0m, "pending")));
Console.WriteLine(ProcessOrder(new Order(2, 150m, "approved")));
Console.WriteLine(ProcessOrder(new Order(3, 150m, "cancelled")));
Console.WriteLine(ProcessOrder(new Order(4, 150m, "unknown")));

string ProcessOrder(Order? order)
{
    if (order is null) return "order is required";
    if (order.Total <= 0) return "total must be greater than zero";

    var result = order.Status switch
    {
        "approved"  => $"Order {order.Id} approved for {order.Total:C}",
        "pending"   => $"Order {order.Id} is pending review",
        "cancelled" => $"Order {order.Id} was cancelled",
        _           => $"Order {order.Id} has unknown status",
    };

    return result;
}

record Order(int Id, decimal Total, string Status);

File: docs/javascript/scripts/control-flow.js
// Princípios: retorno antecipado, fluxo linear, lookup table

const STATUS_LABELS = {
  pending: "Aguardando",
  approved: "Aprovado",
  cancelled: "Cancelado",
};

console.log(processOrder(null));
console.log(processOrder({ items: [], status: "pending" }));
console.log(processOrder({ items: [{ price: 50 }, { price: 30 }], status: "pending" }));

console.log(getStatusLabel("approved"));
console.log(getStatusLabel("unknown"));

function processOrder(order) {
  if (!order) return null;
  if (!order.items.length) return null;
  if (order.status !== "pending") return null;

  const total = order.items.reduce((sum, item) => sum + item.price, 0);
  const processedOrder = { ...order, total, status: "approved" };

  return processedOrder;
}

function getStatusLabel(status) {
  const label = STATUS_LABELS[status] ?? "Desconhecido";
  return label;
}
